pandas/core/categorical.py

```diff
@@ -896,11 +909,41 @@ def rename_categories(self, new_categories, inplace=False):
         remove_categories
         remove_unused_categories
         set_categories
+
+        Examples
+        --------
+        >>> c = Categorical(['a', 'a', 'b'])
+        >>> c.rename_categories([0, 1])
+        [0, 0, 1]
+        Categories (2, int64): [0, 1]
+
+        For dict-like ``new_categories``, extra keys are ignored and
+        categories not in the dictionary are passed through
+
+        >>> c.rename_categories({'a': 'A', 'c': 'C'})
+        [A, A, b]
+        Categories (2, object): [A, b]
+
+        Series are considered list-like here, so the *values* are used
+        instead of the *index*
+
+        >>> c.rename_categories(pd.Series([0, 1], index=['a', 'b']))
+        [0, 0, 1]
+        Categories (2, int64): [0, 1]
         """
         inplace = validate_bool_kwarg(inplace, 'inplace')
         cat = self if inplace else self.copy()
 
-        if is_dict_like(new_categories):
+        is_series = isinstance(new_categories, ABCSeries)
+
+        if is_series:
+            msg = ("Treating Series 'new_categories' as a list-like and using "
+                   "the values. In a future version, 'rename_categories' will "
+                   "treat Series like a dictionary.\n"
+                   "For dict-like, use 'new_categories.to_dict()'\n"
+                   "For list-like, use 'new_categories.values'.")
+            warn(msg, FutureWarning, stacklevel=2)
```

Comment:
Maybe convert the series to array (list-like), so then the rest of the code does not need to take care of it being a series or not

Reply:
> If we go for "Series -> dict-like" behaviour, this is a breaking change, and we need to use a warning for that.

Sorry I think that was an example I added in the first commit of this PR, before we decided to treat Series as list-like.

Reply:
why is this a breaking change at all? we simply did not support this before

Reply:
The old behavior required a list-like, and `Series` are list like. It's not unreasonable for a user to expect

```python
cat.rename(Series([0, 1]))
```

to work, since it did! But we have a new feature that changes the behavior.

Reply:
ok I c, I think this was accidently supported before. ok so fine on the FutureWarning.